### ISSUE TITLE
Device transfer for NJT within torch.compile allocates a new nested int

### ISSUE THREAD

**Issue by jbschlosser** [pytorch/pytorch#137275]
Note that this differs from eager behavior for `njt.to(device)`, where the nested int from the input shape is purposefully includes in the output shape, as expected.

Repro:

```python
import torch

def f(nt):
    return nt.to(device="cpu")

compiled_f = torch.compile(f)

nt = torch.nested.nested_tensor([
    torch.randn(2, 5),
    torch.randn(3, 5),
    torch.randn(4, 5),
], layout=torch.jagged, device="cuda")

out = f(nt)
out_compile = compiled_f(nt)
# fails; torch.Size([7, j1]) for out and torch.Size([7, j2]) for out_compile
assert out.shape == out_compile.shape
```

**Comment by exclamaforte** [comment]
Doing some issue scrubbing: I reproduced this, but the shapes are different now:
```
torch.Size([3, j1, 5]) torch.Size([3, j2, 5])
```
I think no one has picked it up because it has label nested tensor. Any thoughts on who would be a good assignee?

**Comment by jbschlosser** [comment]
The original issue just had the wrong shapes; they were referring to the values buffer of the NJTs. @soulitzer's work would eventually fix all of this, but we have de-prioritized NJT development at the moment in favor of other things. So you can leave this open and perhaps we will return to it in the future.
